test/powershell/engine/Remoting/ImplicitRemotingBatching.Tests.ps1

```diff
@@ -0,0 +1,107 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "TestImplicitRemotingBatching hook should correctly batch simple remote command pipelines" -Tags 'Feature','RequireAdminOnWindows' {
+
+    BeforeAll {
+
+        if ($isWindows)
+        {
+            [powershell] $powerShell = [powershell]::Create([System.Management.Automation.RunspaceMode]::NewRunspace)
+
+            # Create remote session in new PowerShell session
+            $powerShell.AddScript('Import-Module -Name HelpersRemoting; $remoteSession = New-RemoteSession').Invoke()
+            if ($powerShell.Streams.Error.Count -gt 0) { throw "Unable to create remote session for test" }
+
+            # Import implicit commands from remote session
+            $powerShell.Commands.Clear()
+            $powerShell.AddScript('Import-PSSession -Session $remoteSession -CommandName "Get-Process","Write-Output" -AllowClobber').Invoke()
+            if ($powerShell.Streams.Error.Count -gt 0) { throw "Unable to import pssession for test" }
+
+            # Define $filter variable in local session
+            $powerShell.Commands.Clear()
+            $powerShell.AddScript('$filter = "pwsh","powershell"').Invoke()
+            $localRunspace = $powerShell.Runspace
+
+            [powershell] $psInvoke = [powershell]::Create([System.Management.Automation.RunspaceMode]::NewRunspace)
+
+            $testCases = @(
+                @{
+                    Name = 'Two implicit commands should be successfully batched'
+                    CommandLine = 'Get-Process -Name "pwsh" | Write-Output'
+                    ExpectedOutput = $true
+                },
+                @{
+                    Name = 'Two implicit commands with Where-Object should be successfully batched'
+                    CommandLine = 'Get-Process | Write-Output | Where-Object { $_.Name -like "*pwsh*" }'
```

Comment:
What about aliases like `?`

Reply:
I don't know what you mean.  If you have a scenario then let me know.  But I don't intend to cover all scenarios on this initial check in.  This is an experimental feature that will evolve over time.

Reply:
I believe you have code above to see if the command is an alias and convert it to the cmdlet.  So you should just add a test variation where:

```powershell
CommandLine = 'Get-Process | Write-Output | ? { $_.Name -like "*pwsh*" }'
```

Reply:
If we'll use the alias we should add a comment to exclude a unneeded change later.